sdk/forms/src/main/java/com/klaviyo/forms/KlaviyoWebViewDelegate.kt

```diff
@@ -59,102 +68,159 @@ internal class KlaviyoWebViewDelegate : WebViewClient(), WebViewCompat.WebMessag
+     * and initialize klaviyo.js for in-app forms with handshake data injected in the document head
+     */
+    fun initializeWebView() = webView ?: KlaviyoWebView().also { webView ->
+        this.webView = webView
+
+        val klaviyoJsUrl = Uri.parse(Registry.config.baseCdnUrl)
+            .buildUpon()
+            .path("onsite/js/klaviyo.js")
+            .appendQueryParameter("company_id", Registry.config.apiKey)
+            .appendQueryParameter("env", "in-app")
+            .build()
+
+        Registry.config.applicationContext.assets
+            .open("InAppFormsTemplate.html")
+            .bufferedReader()
+            .use(BufferedReader::readText)
+            .replace("SDK_NAME", Registry.config.sdkName)
+            .replace("SDK_VERSION", Registry.config.sdkVersion)
+            .replace("BRIDGE_NAME", this.bridgeName)
+            .replace("BRIDGE_HANDSHAKE", handShakeData)
+            .replace("KLAVIYO_JS_URL", klaviyoJsUrl.toString())
+            .also { html ->
+                webView.loadTemplate(html, this)
+            }
+
+        handshakeTimer?.cancel()
+        handshakeTimer = Registry.clock.schedule(Registry.config.networkTimeout.toLong()) {
+            Registry.log.debug("IAF WebView Aborted: Timeout waiting for Klaviyo.js")
+            close()
         }
     }
 
+    /**
+     * Called when loading a resource encounters http status code >= 400
+     */
     override fun onReceivedHttpError(
         view: WebView?,
         request: WebResourceRequest?,
         errorResponse: WebResourceResponse?
     ) {
-        Registry.log.error("HTTP Error: ${errorResponse?.statusCode} - ${request?.url}")
+        Registry.log.warning("HTTP Error: ${errorResponse?.statusCode} - ${request?.url}")
         super.onReceivedHttpError(view, request, errorResponse)
     }
 
-    override fun shouldInterceptRequest(
-        view: WebView?,
-        request: WebResourceRequest?
-    ): WebResourceResponse? {
-        Registry.log.debug("Request: ${request?.url}")
-        return super.shouldInterceptRequest(view, request)
-    }
-
+    /**
+     * Intercept page navigation and redirect to an external browser application
+     */
     override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
         if (request?.isForMainFrame == true) {
-            Registry.log.debug("Overriding external URL to browser: ${request.url}")
+            Registry.log.info("Redirect URL to external browser: ${request.url}")
             val intent = Intent(Intent.ACTION_VIEW).apply {
                 data = request.url
                 addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
             }
             activity?.let {
                 startActivity(it, intent, null)
             } ?: run {
-                Registry.log.error("Unable to launch external browser, null activity reference")
+                Registry.log.warning("Unable to launch external browser - null activity reference")
```

Comment:
pretty trivial, but `error` and above get output from production builds, so I try to stay at warning or lower unless there's a compelling reason.